frontend/src/helpers/Partie.js

```diff
@@ -1,231 +1,245 @@
+    };
+    // before sending message, reset this turns counters
+    for (const deadPlayer in this.dead_this_turn) {
+      this.dead_players.push(deadPlayer);
+      delete this.living_players[deadPlayer.name];
+    }
+    this.dead_this_turn = [];
+    for (const resurectedPlayer in this.resurected_this_turn) {
+      this.dead_players.remove(resurectedPlayer);
+      this.living_players[resurectedPlayer.name] = resurectedPlayer;
+    }
+    this.resurected_this_turn = [];
+
+    return data;
+  }
+
+  pregame_end() {
+    return this.message('La première nuit vas débuter ...');
+  }
+
+  daytime_end() {
+    // kill with council
+    const playername = this.council.playerToKill();
+    if (!playername) {
+      return this.message("La journée s'achève pour le village");
+    }
+    console.log(
+      'The player ' + playername + ' has been eliminated by the village'
+    );
+    this.council.reset();
+    return this.message("La journée s'achève pour le village");
+  }
+
+  nighttime_end() {
+    const playername = this.council.playerToKill('loups_garou');
+    if (!playername) {
+      return this.message(
+        "La nuit s'achève, le village découvre les actions passé dans la nuit."
+      );
+    }
+    console.log(
+      'The player ' + playername + ' has been eliminated by the werewolfs'
+    );
+    return this.message(
+      "La nuit s'achève, le village découvre les actions passé dans la nuit."
+    );
+  }
+
+  pregameActions(who, vote, extra = {}) {}
+
+  daytime_action(who, vote, extra = {}) {
+    const player = this.livingPlayers[this.playing];
+    if (who !== player.playername) {
+      throw new Error('Player trying to play is not the current player');
+    }
+    player.actionDay(this.council, vote, extra);
+    return this.nextPlayer();
+  }
+
+  nighttime_action(who, vote, extra = {}) {
+    const player = this.livingPlayers[this.playing];
+    if (who !== player.playername) {
+      throw new Error('Player trying to play is not the current player');
+    }
+    extra['dead_this_turn'] = this.dead_this_turn;
+    this.dead_this_turn = player.nighttime_action();
+  }
+
+  action(who, vote, extra = {}) {
+    const player = this.livingPlayers[this.playing];
+    if (who !== player.playername) {
+      throw new Error('Player trying to play is not the current player');
+    }
+    let data = {};
+    if (this.phase === Phases.Day) {
+      data = player.actionDay(this.council, vote, extra);
+    }
+    if (this.phase === Phases.Night) {
+      data = player.actionNight(this.council, vote, extra);
+    }
+    if (this.phase === Phases.PreGame) {
+      data = player.actionPregame(this.council, vote, extra);
+    }
+    this.cycle.shift();
+    return data;
+  }
+  printPlayers() {
+    console.log('Affichage des joueurs et leurs rôles:');
+    for (let i = 0; i < this.players.length; i++) {
+      console.log(
+        `Nom du Joueur  ${i}: Rôle du Joueur ${this.players[i].role.role}`
```

Comment:
this.players[i].role.role devrait juste être this.players[i].role, si ya besoin d'un .role en plus c'est que ya un probleme quelque part